src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateCopy = { ...state };
+  const stateVersions = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          delete stateCopy[property];
+        }
```

Comment:
could be separate function

Reply:
like this?
`function addProperties(extraData, state, stateVersions) {
  for (const property in extraData) {
    state[property] = extraData[property];
    }
     stateVersions.push({ ...state });
  }`
  
  I was thinking abt separate functions, but decided that it`s too much code, so I prefer shorter solution here.

Reply:
@PANCH1811 solution should be **_readable_**, not shorter